src/test/java/seedu/address/logic/NewLogicManagerTest.java

```diff
@@ -0,0 +1,173 @@
+package seedu.address.logic;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static seedu.address.logic.Messages.MESSAGE_INVALID_DOCTOR_DISPLAYED_INDEX;
+import static seedu.address.logic.Messages.MESSAGE_UNKNOWN_COMMAND;
+import static seedu.address.logic.commands.CommandTestUtil.NAME_DESC_AMY;
+import static seedu.address.logic.commands.CommandTestUtil.NRIC_DESC_AMY;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalDoctor.AMY;
+
+import java.io.IOException;
+import java.nio.file.AccessDeniedException;
+import java.nio.file.Path;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import seedu.address.logic.commands.AddDoctorCommand;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.ListDoctorCommand;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.NewModel;
+import seedu.address.model.NewModelManager;
+import seedu.address.model.ReadOnlyDatabase;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.person.doctor.Doctor;
+import seedu.address.storage.JsonDatabaseStorage;
+import seedu.address.storage.JsonUserPrefsStorage;
+import seedu.address.storage.NewStorageManager;
+import seedu.address.testutil.DoctorBuilder;
+
+public class NewLogicManagerTest {
+    private static final IOException DUMMY_IO_EXCEPTION = new IOException("dummy IO exception");
+    private static final IOException DUMMY_AD_EXCEPTION = new AccessDeniedException("dummy access denied exception");
+
+    @TempDir
+    public Path temporaryFolder;
+
+    private NewModel newModel = new NewModelManager();
+    private NewLogic newLogic;
+
+    @BeforeEach
+    public void setUp() {
+        JsonUserPrefsStorage userPrefsStorage = new JsonUserPrefsStorage(temporaryFolder.resolve("userPrefs.json"));
+        JsonDatabaseStorage databaseStorage =
+                new JsonDatabaseStorage(temporaryFolder.resolve("addressBook.json"));
```

Comment:
should be "database.json" instead of "addressBook.json"

Reply:
done